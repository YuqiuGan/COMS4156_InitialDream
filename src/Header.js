import React from 'react'
import './Header.css'
import { Link } from 'react-router-dom'
import SearchIcon from '@material-ui/icons/Search'
import ShoppingBasketIcon from '@material-ui/icons/ShoppingBasket'
import { auth } from './firebase'
import { useStateValue } from './StateProvider'

function Header () {
  const [{ basket, user }] = useStateValue()
  // console.log(basket);

  const handleLogin = () => {
    if (user) {
      auth.signOut()
    }
  }

  return (
    <div className='header'>
      <Link to='/'>
        <img
          className='header__logo'
          src='CU_logo.png'
          alt=''
        />
      </Link>

      <div className='header_search'>
        <input type='text' className='header__searchInput' />
        <SearchIcon className='header__searchIcon' />
      </div>

      <div className='header__nav'>
        <Link to={!user && '/login'} className='header__link'>
          <div onClick={handleLogin} className='header__option'>
            <span className='header__optionLineOne'>Hello {!user ? 'Guest' : user.email}</span>
            <span className='header__optionLineTwo'>{user ? 'Sign Out' : 'Sign In'}</span>
          </div>
        </Link>
        {/* 2nd link */}
        <Link to='/introduction' className='header__link'>
          <div className='header__option'>
            <span className='header__optionLineOne'>Health Code</span>
            <span className='header__optionLineTwo'>FAQ</span>
          </div>
        </Link>
        {/* 3rd link */}
        <Link className='header__link'>
          <div className='header__option'>
            <span className='header__optionLineOne'>Your</span>
            <span className='header__optionLineTwo'>Profile</span>
          </div>
        </Link>
        {/* 4th link */}
        <Link to='/checkout' className='header__link'>
          <div className='header__optionBasket'>
            {/* bascket icon */}
            <ShoppingBasketIcon />
            {/* Number of items */}
            <span className='header__optionLineTwo header__basket'>{basket.length}</span>
          </div>
        </Link>
      </div>
    </div>
  )
}

export default Header
