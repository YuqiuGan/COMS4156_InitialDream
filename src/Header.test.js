import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import Header from './Header'
import Login from './Login'
import { create } from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <Header />
    </BrowserRouter>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})

describe(' Simple Snapshot Test', () => {
  test('Testing square', () => {
    const tree = create(<BrowserRouter><Login /></BrowserRouter>)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})

it('onClick is called when button is clicked', () => {
  const fn = jest.fn()
  const tree = create(<BrowserRouter> <Login onClick={fn} /></BrowserRouter>)
  // Simulate button click
  const button = tree.root.findByType('button')
  button.props.onClick()
  // Verify callback is invoked
  expect(fn.mock.calls.length).toBe(1)
})
