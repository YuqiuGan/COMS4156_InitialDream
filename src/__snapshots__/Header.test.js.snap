// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Simple Snapshot Test Testing square 1`] = `
<div
  className="login"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt=""
      className="login__logo"
      src="CU_logo.png"
    />
  </a>
  <div
    className="login__container"
  >
    <h1>
      Sign-in
    </h1>
    <form>
      <h5>
        E-mail
      </h5>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <h5>
        Password
      </h5>
      <input
        onChange={[Function]}
        type="password"
        value=""
      />
      <button
        className="login__signInButton"
        onClick={[Function]}
        type="submit"
      >
        Sign In
      </button>
    </form>
    <p>
      By signing-in you agree to the AMAZON FAKE CLONE Conditions of Use & Sale. Please see our Privacy Notice, our Cookies Notice and our Interest-Based Ads Notice.
    </p>
    <button
      className="login__registerButton"
      onClick={[Function]}
    >
      Create your Amazon Account
    </button>
  </div>
</div>
`;
